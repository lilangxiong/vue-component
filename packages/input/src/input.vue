<template>
  <div>
    <input v-bind="$attrs" :type="type" :value="value" @input="handleInput">
  </div>
</template>

<script>
export default {
  name: 'lgInput',
  inheritAttrs: false, // 不让div继承属性，通过v-bind="$attrs"实现指定继承
  props: {
    value: {
      type: String,
      default: ''
    },
    type: {
      type: String,
      default: 'text'
    }
  },
  methods: {
    handleInput(evt) {
      this.$emit('input', evt.target.value)

      // 不推荐使用依赖注入，因为input组件可以被单独使用，此时是没有form-item组件的
      const findParent = (parent) => {
        while(parent) {
          if (parent.$options.name === 'lgFormItem') {
            break
          } else {
            parent = parent.$parent
          }
        }
        return parent
      }

      const parent = findParent(this.$parent)
      if (parent) parent.$emit('validate')
    }
  }
}
</script>

<style lang='scss' scoped>
</style>
