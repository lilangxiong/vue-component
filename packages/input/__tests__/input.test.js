import input from '../src/input.vue'
import { mount } from '@vue/test-utils'

describe('llx-input', () => {
  test('input-text', () => {
    const wrapper = mount(input)
    expect(wrapper.html()).toContain('input type="text"')
  })

  test('input-text', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.html()).toContain('input type="password"')
  })

  test('input-text', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: 'admin'
      }
    })
    expect(wrapper.props('value')).toBe('admin')
  })

  test('input-sanpshot', () => {
    const wrapper = mount(input, {
      propsData: {
        type: 'password',
        value: ''
      }
    })
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})