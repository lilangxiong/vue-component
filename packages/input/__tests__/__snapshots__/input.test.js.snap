// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`llx-input input-sanpshot 1`] = `
<div>
  <input
    type="password"
  />
</div>
`;
